cli/worker.rs

```diff
@@ -126,13 +126,18 @@ impl CliMainWorker {
         self.pending_unload = true;
 
         let result = loop {
-          let result = self.inner.worker.run_event_loop(false).await;
-          if !self
+          match self.inner.worker.run_event_loop(false).await {
+            Ok(()) => {}
+            Err(error) => break Err(error),
+          }
+          match self
             .inner
             .worker
-            .dispatch_beforeunload_event(&located_script_name!())?
+            .dispatch_beforeunload_event(&located_script_name!())
           {
-            break result;
+            Ok(default_prevented) if default_prevented => {} // continue loop
+            Ok(_) => break Ok(()),
+            Err(error) => break Err(error),
```

Comment:
The unload on drop case is only meant to be hit when the watcher restarts on file changes mid-execution, but due to faulty circuiting here it could be hit on `"beforeunload"` handler errors or other error events (#15428). It's fixed with this